filter_plugins/oo_filters.py

```diff
@@ -491,6 +494,95 @@ def to_padded_yaml(data, level=0, indent=2, **kw):
         except Exception as my_e:
             raise errors.AnsibleFilterError('Failed to convert: %s', my_e)
 
+    @staticmethod
+    def oo_openshift_env(hostvars):
+        ''' Return facts which begin with "openshift_"
+            Ex: hostvars = {'openshift_fact': 42,
+                            'theyre_taking_the_hobbits_to': 'isengard'}
+                returns  = {'openshift_fact': 42}
+        '''
+        if not issubclass(type(hostvars), dict):
+            raise errors.AnsibleFilterError("|failed expects hostvars is a dict")
+
+        facts = {}
+        regex = re.compile('^openshift_.*')
+        for key in hostvars:
+            if regex.match(key):
+                facts[key] = hostvars[key]
+        return facts
+
+    @staticmethod
+    # pylint: disable=too-many-branches
+    def oo_persistent_volumes(hostvars, groups, persistent_volumes=None):
+        """ Generate list of persistent volumes based on oo_openshift_env
+            storage options set in host variables.
+        """
+        if not issubclass(type(hostvars), dict):
+            raise errors.AnsibleFilterError("|failed expects hostvars is a dict")
+        if not issubclass(type(groups), dict):
+            raise errors.AnsibleFilterError("|failed expects groups is a dict")
+        if persistent_volumes != None and not issubclass(type(persistent_volumes), list):
+            raise errors.AnsibleFilterError("|failed expects persistent_volumes is a list")
+
+        if persistent_volumes == None:
+            persistent_volumes = []
+        for component in hostvars['openshift']['hosted']:
+            kind = hostvars['openshift']['hosted'][component]['storage']['kind']
+            create_pv = hostvars['openshift']['hosted'][component]['storage']['create_pv']
+            if kind != None and create_pv:
+                if kind == 'nfs':
+                    host = hostvars['openshift']['hosted'][component]['storage']['host']
+                    if host == None and len(groups['oo_nfs_to_config']) > 0:
+                        host = groups['oo_nfs_to_config'][0]
+                    else:
+                        raise errors.AnsibleFilterError("|failed no storage host detected")
```

Comment:
No host was configured as an override and no `[nfs]` group was configured so we can't make the PV.